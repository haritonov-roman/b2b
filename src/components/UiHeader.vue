<script>
export default {
  name: 'UiHeader',

  computed: {
    rep () {
      return this.$route.params.rep ? `: ${this.$route.params.rep}` : ''
    }
  }
}
</script>

<template>
  <cds-header-service>
    <template #left>
      <div class="header">
        <div class="header__logo">
          <img src="@/assets/svg/github-mark.svg" alt="github-mark" />
        </div>
        <h1 class="header__title">GitHub issue viewer{{ rep }}</h1>
      </div>
    </template>
  </cds-header-service>
</template>

<style lang="scss" scoped>
@import "@central-design-system/components/dist/mixins/scss/layout/convert";

.header {
  display: flex;
  grid-column-gap: toRem(16px);
  align-items: center;

  &__logo {
    width: toRem(32px);
    height: toRem(32px);
  }

  &__title {
    font-size: var(--cds-font-heading-4);
  }
}
</style>
