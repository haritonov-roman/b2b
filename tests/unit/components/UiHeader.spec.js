import { createLocalVue, shallowMount } from '@vue/test-utils'
import Cds from '@central-design-system/components'
import UiHeader from '@/components/UiHeader'

const localVue = createLocalVue()

localVue.use(Cds)

const $route = {
  params: {
    rep: 'vue'
  }
}

describe('UiHeader.vue', () => {
  const wrapper = shallowMount(UiHeader, {
    mocks: {
      $route
    },
    localVue
  })

  it('Проверка рендеринга', () => {
    expect(wrapper.exists()).toBe(true)
  })

  it('Проверка содержимого', () => {
    expect(wrapper.text()).toMatch('GitHub issue viewer: vue')
  })

  it('Снимок', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
