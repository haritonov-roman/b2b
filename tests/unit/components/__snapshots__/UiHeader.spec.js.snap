// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UiHeader.vue Снимок 1`] = `
<cds-header-service-stub>
  <div
    class="header"
  >
    <div
      class="header__logo"
    >
      <img
        alt="github-mark"
        src="@/assets/svg/github-mark.svg"
      />
    </div>
     
    <h1
      class="header__title"
    >
      GitHub issue viewer: vue
    </h1>
  </div>
</cds-header-service-stub>
`;
